### PR TITLE
Feature documentation

### PR DESCRIPTION
**Pull request by moacir-ramos-movile**
- Added auto generated documentation page
- Included javadocs to documentation page
- Improved javadocs for model classes

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,25 +1,26 @@
...
 ```gradle
 dependencies {
   compile 'com.movile.botland:botland-sdk-java:+'
 }
 ```
 
-## Examples
+## Usage
+
+This SDK allows you to connect do Botland and explore all the APIs. Please visit [Botland Documentation](http://botland.io/manager/#documentation/getting-started "Botland Documentation") to learn how to get your credentials and create your ```citizenId```.
```
Comment (rodrigo-marchesini-movile) [inline README.md]: typo `connect do Botland`

File: README.md [hunk 2]
```diff
@@ -79,9 +80,11 @@ Check this echo bot example.
           .start();
 
 ```
-### Other examples
+### More examples
 
-See [examples](src/main/java/com/movile/botland/sdk/example).
+Please access the link below to see more examples and explore botland features.
```
Comment (rodrigo-marchesini-movile) [inline README.md]: How about always writing Botland with a capital B?

File: src/main/java/com/movile/botland/sdk/model/Bot.java [hunk 3]
```diff
@@ -27,50 +30,100 @@
...
+     * Get bot id
+     *
+     * @return id
+     */
     public String getId() {
         return id;
     }
 
+    /**
+     * Get id
```
Comment (rodrigo-marchesini-movile) [inline src/main/java/com/movile/botland/sdk/model/Bot.java]: typo Set

File: src/main/java/com/movile/botland/sdk/model/Contact.java [hunk 4]
```diff
@@ -33,98 +38,194 @@
...
+    /**
+     * Contact id from the original messenger platform
+     * @return externalId
+     */
     public String getExternalId() {
         return externalId;
     }
 
+    /**
+     * Get external id
```
Comment (rodrigo-marchesini-movile) [inline src/main/java/com/movile/botland/sdk/model/Contact.java]: typo: Set

File: src/main/java/com/movile/botland/sdk/model/Conversation.java [hunk 5]
```diff
@@ -19,93 +19,195 @@
...
+     * Get original messenger platform
+     *
+     * @return platform
+     */
     public String getPlatform() {
         return platform;
     }
 
+    /**
+     * Get platform
```
Comment (rodrigo-marchesini-movile) [inline src/main/java/com/movile/botland/sdk/model/Conversation.java]: Typo: Set

File: src/main/java/com/movile/botland/sdk/model/MessageType.java [hunk 6]
```diff
@@ -19,6 +19,9 @@
 import java.util.Arrays;
 import java.util.Optional;
 
+/**
+ * This class enumerate available message types is Botland.
```
Comment (rodrigo-marchesini-movile) [inline src/main/java/com/movile/botland/sdk/model/MessageType.java]: typo: enumerates

File: src/main/java/com/movile/botland/sdk/model/MessageType.java [hunk 7]
```diff
@@ -37,10 +40,21 @@ private MessageType(String value) {
...
+     * Get message type value
+     * @return value
+     */
     public String getValue() {
         return value;
     }
 
+    /**
+     * This method search a message type and return an enum item or
+     * UNKNOWN is it not possible to identify the message type.
```
Comment (rodrigo-marchesini-movile) [inline src/main/java/com/movile/botland/sdk/model/MessageType.java]: typo: if

**Comment by rodrigo-marchesini-movile** [comment]
I reviewed the text. It would be nice if @marcelo-alves-movile could check the resulting website ;)

**Comment by marcelo-alves-movile** [comment]
LGTM

**Pull request merged**